experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts

```diff
@@ -111,6 +123,16 @@ export interface HasReattachFields extends HasPlaceFields {
 	 */
 	inverseOf?: RevisionTag;
 }
+// TODO
+export const HasReattachFields = Type.Intersect([
+	HasPlaceFields,
+	Type.Object({
+		detachedBy: Type.Optional(RevisionTagSchema),
+		detachIndex: Type.Number(),
+		isIntention: OptionalTrue,
+		lastDetachedBy: Type.Optional(RevisionTagSchema),
+	}),
+]);
```

Comment:
Did you forget about this or is there a compelling reason to leave it in this state?
I'm guessing it should be:
```suggestion
export const HasReattachFields = Type.Intersect([
	HasPlaceFields,
	Type.Object({
		inverseOf: Type.Optional(RevisionTagSchema),
	}),
]);
```

Reply:
Also, I'm surprised this doesn't lead to a failure.

Reply:
thank you, bad merge :).

The schemas here aren't used yet. I'm breaking up #15237 into more reviewable chunks. Once that is all in, this would have definitely caused test failures.

Reply:
specifically, my current plan is to make `IJsonCodec` optionally supply a schema that its data should match. Generally, top-level codecs will build their expected schema based on those and verify all of the encoded data in a single pass. But alternative patterns are possible--defining a single schema for our tree types gets pretty complex since it's mutually recursive. it's possible, but I think just verifying 'layers' of the tree independently is simpler approach that is still pretty easy to write.

For this case (fields), my plan is to verify schema in `modularChangeCodecs.ts`.